### DISCUSSION TITLE
401 Unauthorized spam

### DISCUSSION THREAD

**Discussion by SpasiboKojima** [WalletConnect/walletconnect-monorepo#741] [Developer Support] [answered]
My Dapp supports only BSC networks and it works fine, but when I try to scan the QR while being connected to ETH main chain it starts spamming requests to infura with 401 Unauthorized responses. I didn't provide it in `rpc` field at all and have only `56` and `97` networks there. The spam doesn't stop no matter what I do, refresh is the only thing that gets it away.
Can be fixed by disconnecting from wallet as it appears, but still the behavior is weird to me.
![Screenshot from 2022-01-18 14-55-38](https://user-images.githubusercontent.com/34808650/149933578-ad9fe22c-9fd9-4981-b59e-b79494d05e91.png)

**Comment by dalechyn** [comment, marked as answer]
You didn't provide matching for `chainId:1` rpc in `rpc` object so it tries to connect to infura.
Disable walletconnect button if the user is not connected to bsc networks, a common practice

**Comment by SaifulJnU** [comment]
Solved !!!
Just go for the following link:
https://github.com/SaifulJnU/IFPS_Template/blob/master/IPFS%20infura%20unauthorized%20problem.txt
